service/matching/service.go

```diff
@@ -111,19 +112,40 @@ func (s *Service) Start() {
 // Stop stops the service
 func (s *Service) Stop() {
 	// remove self from membership ring and wait for traffic to drain
-	s.logger.Info("ShutdownHandler: Evicting self from membership ring")
-	if err := s.membershipMonitor.EvictSelf(); err != nil {
+	var err error
+	var waitTime time.Duration
+	if align := s.config.AlignMembershipChange(); align > 0 {
+		propagation := s.membershipMonitor.ApproximateMaxPropagationTime()
+		asOf := util.NextAlignedTime(time.Now().Add(propagation), align)
+		s.logger.Info("ShutdownHandler: Evicting self from membership ring as of", tag.Timestamp(asOf))
+		waitTime, err = s.membershipMonitor.EvictSelfAt(asOf)
+	} else {
+		s.logger.Info("ShutdownHandler: Evicting self from membership ring immediately")
+		err = s.membershipMonitor.EvictSelf()
+	}
+	if err != nil {
 		s.logger.Error("ShutdownHandler: Failed to evict self from membership ring", tag.Error(err))
 	}
 	s.healthServer.SetServingStatus(serviceName, healthpb.HealthCheckResponse_NOT_SERVING)
```

Comment:
It can be extracted, yes, but I don't see what difference it makes with the order of this line.

Btw the health check status is not used for anything meaningful here (well, maybe k8s will kill the pod eventually but we'll be done before that happens)

Reply:
What part exactly do you want to extract and why? I'm trying to figure out a rearrangement that makes it easier to read and I can't see it